server/controllers/stock/reports/index.js

```diff
@@ -7,868 +7,18 @@
-      bundle.delay = options.inventory_delay;
-      bundle.purchaseInterval = options.purchase_interval;
-    }
-
-    report = new ReportManager(STOCK_INVENTORIES_REPORT_TEMPLATE, req.session, optionReport);
-  } catch (e) {
-    return next(e);
-  }
-
-  return Stock.getInventoryQuantityAndConsumption(options)
-    .then((rows) => {
-      data.rows = rows;
-      data.hasFilter = hasFilter;
-      data.filters = filters;
-      data.csv = rows;
-      data.display = display;
-
-      data.dateTo = options.dateTo;
-      data.delay = bundle.delay || 1;
-      data.purchaseInterval = bundle.purchaseInterval || 1;
-
-      // group by depot
-      let depots = _.groupBy(rows, d => d.depot_text);
-
-      // make sure that they keys are sorted in alphabetical order
-      depots = _.mapValues(depots, lines => {
-        _.sortBy(lines, 'depot_text');
-        return lines;
-      });
-
-      data.depots = depots;
-      return report.render(data);
-    })
-    .then((result) => {
-      res.set(result.headers).send(result.report);
-    })
-    .catch(next)
-    .done();
-}
-
-/**
- * @method stockInventoryReport
- *
- * @description
- * This method builds the stock inventory report as either a JSON, PDF, or HTML
- * file to be sent to the client.
- *
- * GET /reports/stock/inventory
- */
-function stockInventoryReport(req, res, next) {
-  const data = {};
-  let options;
-  let report;
-
-  const optionReport = _.extend(req.query, {
-    filename : 'TREE.STOCK_INVENTORY_REPORT',
-    orientation : 'landscape',
-    footerRight : '[page] / [toPage]',
-    footerFontSize : '8',
-  });
-
-  // set up the report with report manager
-  try {
-    options = req.query.params ? JSON.parse(req.query.params) : {};
-    report = new ReportManager(STOCK_INVENTORY_REPORT_TEMPLATE, req.session, optionReport);
-  } catch (e) {
-    return next(e);
-  }
-
-  return db.one('SELECT code, text FROM inventory WHERE uuid = ?;', [db.bid(options.inventory_uuid)])
-    .then((inventory) => {
-      data.inventory = inventory;
-
-      return db.one('SELECT text FROM depot WHERE uuid = ?;', [db.bid(options.depot_uuid)]);
-    })
-    .then((depot) => {
-      data.depot = depot;
-
-      return Stock.getInventoryMovements(options);
-    })
-    .then((rows) => {
-      data.rows = rows.movements;
-      data.totals = rows.totals;
-      data.result = rows.result;
-      data.csv = rows.movements;
-      data.dateTo = options.dateTo;
-
-      return report.render(data);
-    })
-    .then((result) => {
-      res.set(result.headers).send(result.report);
-    })
-    .catch(next)
-    .done();
-}
+const stockLotsReport = require('./stock/lots_report');
+const stockMovementsReport = require('./stock/movements_report');
+const stockInventoriesReport = require('./stock/inventories_report');
+const stockInventoryReport = require('./stock/inventory_report');
+const stockExitPatientReceipt = require('./stock/exit_patient_receipt');
```

Comment:
It looks like you have a typo in the file name which makes this `require()` fail.  The file is called `exit_patent_receipt` instead of `exit_patient_receipt`.

Reply:
Ah great thanks for the heads up. I see the travis page now - I will commit and check the output make sure there isn't another typo.

Reply:
Ok I can see a different error now regarding Route.get:
https://travis-ci.org/IMA-WorldHealth/bhima-2.X/jobs/273610354

Reply:
Sweet!   This means that there is an `app.get()` somewhere that isn't being given a function.  We'll have to track that down...

Reply:
All the app.get( lines for stockReports seems to be okay in terms of referencing existing functions. I can see it is failing on this line in travis: server/config/routes.js:583, which is the first line like this:
```
app.get('/reports/stock/lots', stockReports.stockLotsReport);
```

It is the same exports as before so shouldn't be an issue with that:
https://github.com/jambroo/bhima-2.X/blob/89dbd443b3a5bbdfde386ab5f5cf2f97343e924e/server/controllers/stock/reports/index.js#L24

Does anything look wrong to you there?

Reply:
Nothing looks wrong online - let me pull your branch and play around with it.  I'll report back when I'm able to figure something out!

Reply:
Aha - gotcha.  Each of the `require()`s are nested one to deep.  In other words, you export the functions using `exports.fn = fn`, and then pass them up to the top via another `exports.fn = fn`.  Specifically, each of [these lines](https://github.com/IMA-WorldHealth/bhima-2.X/pull/2084/files#diff-a51ce60ba2265ea47425f240a42b96cdL874) should either read:
```js
 exports.stockLotsReport = stockLotsReport.stockLotsReport;
// etc ...
```
OR

You can change [the exports line to using module.exports](https://github.com/IMA-WorldHealth/bhima-2.X/pull/2084/files#diff-37cc91ea7119fcc956060bf5ae71be36R70).

I've gone ahead and fixed this while I was messing around with you code base.  I think it should all pass - let's see if Travis is happy with it!